### PR TITLE
Rewrite in TypeScript, Add method toString, Attribute innerHTML\ oute…

### PR DESCRIPTION
**Pull request by taoqf**
…rHTML

### CODE REVIEW

**Comment by taoqf** [comment]
@ashi009 are you still maintaining this project?

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,5 @@ bower_components
 !.gitignore
 
 *.sublime-*
+dist/
+yarn.lock
```
Comment (ashi009) [inline .gitignore]: Tailing \n.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			"./tests/"
+		],
+		"allowJs": false,
+		"allowUnreachableCode": false,
+		"allowUnusedLabels": false,
+		"alwaysStrict": true,
+		"baseUrl": "",
+		"charset": "utf8",
+		"declaration": true,
+		// "declarationDir": "./dts/",
```
Comment (ashi009) [inline tsconfig.json]: Please drop commented lines from this JSON file. Is this conventional for tsconfig?

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+	text: string;
+	rawText: string;
+	abstract toString(): String;
+}
+/**
+ * TextNode to contain a text element in DOM tree.
+ * @param {string} value [description]
+ */
+export class TextNode extends Node {
+	constructor(value: string) {
```
Comment (ashi009) [inline src/index.ts]: Nice touch of TS. Finally dropped apollo dependency.

**Comment by ashi009** [comment]
This looks good.  Let me know if you still want to merge this.  I'll make you a contributor, so you can maintain this if you want.

**Pull request closed without merging**